src/components/nimbleDataTable/NimbleDataTable.tsx

```diff
@@ -151,11 +153,15 @@ export const NimbleDataTable: React.FC<NimbleDataTableProps> = ({
   const filterChangeDebounceHandler = useMemo(() => debounce(handleFilterChange, 500), [filterData]);
 
   const handleClicShort = (sortKey: string | undefined, sortOrder: string): void => {
-    if (sortKey) {
-      setSortData({
-        ...(isEnableMultipleSort ? sortData : {}),
-        [sortKey]: sortOrder,
-      });
+    if (onClickSort && typeof onClickSort === 'function') {
```

Comment:
If a sorting callback is provided, then the callback is executed.
Otherwise, fallback to existing FE sorting for backward compatibility.